bundles/org.palladiosimulator.probeframework/src/org/palladiosimulator/probeframework/calculator/ICalculatorFactoryLegacyAdapter.java

```diff
@@ -0,0 +1,176 @@
+import static org.palladiosimulator.metricspec.constants.MetricDescriptionConstants.RECONFIGURATION_TIME_METRIC_TUPLE;
+import static org.palladiosimulator.metricspec.constants.MetricDescriptionConstants.RESOURCE_DEMAND_METRIC_TUPLE;
+import static org.palladiosimulator.metricspec.constants.MetricDescriptionConstants.RESPONSE_TIME_METRIC_TUPLE;
+import static org.palladiosimulator.metricspec.constants.MetricDescriptionConstants.STATE_OF_ACTIVE_RESOURCE_METRIC_TUPLE;
+import static org.palladiosimulator.metricspec.constants.MetricDescriptionConstants.STATE_OF_PASSIVE_RESOURCE_METRIC_TUPLE;
+import static org.palladiosimulator.metricspec.constants.MetricDescriptionConstants.WAITING_TIME_METRIC_TUPLE;
+
+import java.util.List;
+
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPoint;
+import org.palladiosimulator.probeframework.probes.Probe;
+import org.palladiosimulator.probeframework.probes.ProbeConfigurations;
+
+/**
+ * This interface provides default implementations to adapt the legacy
+ * ICalculatorFactory to the new interface IGenericCalculatorFactory.
+ * 
+ * When developing new code please use <code>IGenericCalculatorFactory</code>
+ * directly.
+ *
+ * @author Sebastian Krach
+ */
+@SuppressWarnings("deprecation")
+public interface ICalculatorFactoryLegacyAdapter extends ICalculatorFactory, IGenericCalculatorFactory {
+    /**
+     * This factory method allows to create an adapter which delegates the legacy
+     * factory methods to the extensible one.
+     * 
+     * @param delegate the factory which adheres to the new style.
+     * @return the delegating adapter
+     */
+    public static ICalculatorFactoryLegacyAdapter createDelegatingAdapter(final IGenericCalculatorFactory delegate) {
+        return delegate::buildCalculator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    default Calculator buildResponseTimeCalculator(final MeasuringPoint measuringPoint, final List<Probe> probes) {
+        return buildCalculator(RESPONSE_TIME_METRIC_TUPLE, measuringPoint,
+                ProbeConfigurations.createStartStopProbeConfiguration(probes.get(0), probes.get(1)));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    default Calculator buildDemandBasedWaitingTimeCalculator(final MeasuringPoint measuringPoint,
+            final List<Probe> probes) {
+        /**
+         * This kind of calculator cannot be mapped to the new interface as it uses the
+         * same metric description as the standard waiting time calculator. Since the
+         * metric is the same, at least in SimuLizar there is no way of selecting this
+         * calculator. Since calculators were already cached based on their metric
+         * description both methods were already ambiguous. If the differentiation is
+         * required, an additional metric should be defined. [Sebastian Krach]
+         */
+        return buildCalculator(WAITING_TIME_METRIC_TUPLE, measuringPoint,
```

Comment:
You could stress this even more by delegating to the other method providing the same metric. In addition, it is easier to maintain.